kite-data/kite-data-core/src/main/java/org/kitesdk/data/spi/filesystem/FileSystemView.java

```diff
@@ -214,6 +215,24 @@ public long getLastModified() {
         throw new DatasetIOException("Cannot find last modified time of of " + dir, e);
       }
     }
+
+    // if view was marked ready more recently count it as the modified time
+    long readyTimestamp = ((FileSystemDataset<E>)dataset).getReadyTimestamp(getConstraints());
+    if(lastMod < readyTimestamp) {
+      lastMod = readyTimestamp;
+    }
+
     return lastMod;
   }
+
+  @Override
+  public void signalReady() {
+    ((FileSystemDataset<E>)dataset).signalReady(getConstraints());
```

Comment:
I tried both routes and I think injecting the SignalManager into the FileSystemView on construction (similar to partition listener) is the better route.  The AbstractRefinableView route had the issue that the SignalManager itself is an implementation, so if HBase decided to go a different route than the FileSystem storage it would be awkward at best. Going the constructor route hides any of the details of how the signals are managed.

820d9c8f